lib/em/protocols/smtpclient.rb

```diff
@@ -321,7 +321,7 @@ def receive_rcpt_to_response
       end
 
       def escape_leading_dots(s)
-        s.gsub(/^\.([^.]|$)/) { "..#{$1}" }
+        s.gsub(/^\.(.|$)/) { "..#{$1}" }
```

Comment:
Yes, because we need to reinsert the character caught by the wildcard `.` inside the capture group in the case that the leading dot is not the only character on the line.